src/nvim/ex_docmd.c

```diff
@@ -9411,3 +9411,36 @@ static void ex_terminal(exarg_T *eap)
     xfree(name);
   }
 }
+
+static void ex_plug(exarg_T *eap)
+{
+  char *path = NULL;
+  if (*eap->arg != NUL) {
+    if (os_isdir(eap->arg)) {  // full path: plug ~/.config/nvim/plugged/coolplugin
+      path = (char *)vim_strsave_fnameescape(eap->arg, FALSE);
+    } else {  // spec: plug neovim/neovimplugin
+      char_u *user_vim = vim_strsave(p_rtp);
+      strtok((char *)user_vim, ","); // split(&runtimepath, ',')[0]
+      char_u *basename = path_tail((char_u *)eap->arg); // 'neovim/neovimplugin' -> 'neovimplugin'
```

Comment:
Also a variant when `:plug` *only* calls an autocmd event looks more promising. Just add an early plugin which defines the standard event which messes with `&runtimepath` on its own, and document that plugin managers should do

    augroup PluginManager
        execute 'autocmd!' | " Clear autocommands defined by NeoVim early plugin.
        autocmd PluginRequested * :call my_plugin_manager#add_plugin([expand('<amatch>'])
    augroup END